### ISSUE TITLE
`05-led-roulette` errors when building

### ISSUE THREAD

**Issue by alex-schulster** [rust-embedded/discovery#461]
When trying to build the `05-led-roulette` project, I get errors:

```
...
error[E0658]: trait bounds other than `Sized` on const fn parameters are unstable
   --> /Users/alex/.cargo/registry/src/github.com-1ecc6299db9ec823/fixed-1.16.0/src/macros_const.rs:25:14
    |
25  |           impl<Frac: Unsigned> $Fixed<Frac> {
    |                ^^^^
26  |               const fn from_const<SrcFrac: Unsigned>(src: FixedU128<SrcFrac>) -> $Fixed<Frac> {
    |               ------------------------------------------------------------------------------- function declared as const here
    |
   ::: /Users/alex/.cargo/registry/src/github.com-1ecc6299db9ec823/fixed-1.16.0/src/lib.rs:657:1
    |
657 | / fixed! {
658 | |     "A 128",
659 | |     FixedI128(i128, LeEqU128, U128, U127, "128", "129", "127", "126", "125", "124"),
660 | |     16, "0x1234_5678_9ABC_DEF0_0102_0304_0506_0708",
...   |
668 | |     False, "128", FixedI128, i128, FixedI128, i128
669 | | }
    | |_- in this macro invocation
    |
    = note: see issue #93706 <https://github.com/rust-lang/rust/issues/93706> for more information
    = note: this error originates in the macro `fixed_const` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0658`.
error: could not compile `fixed` due to 666 previous errors
```

All the 666 errors are the same `error[E0658]: trait bounds other than 'Sized' on const fn parameters are unstable` error, but occurring in different places.

I use:
* rustc 1.60.0 (7737e0b5c 2022-04-04)
* cargo-size 0.3.3
* cargo-embed 0.11.0

Idk if this is linked to this problem, but I also get an error in the `main.rs` file:

<img width="434" alt="image" src="https://user-images.githubusercontent.com/40316586/178334715-6b44a819-dca5-435b-9792-9cb98ac16f4d.png">

**Comment by alex-schulster** [comment]
Exactly the same command. Maybe I should have precised, I am on macOS.
```
~/Local documents/Code/Rust/embedded/microbit/src/05-led-roulette % cargo build --features v2 --target thumbv7em-none-eabihf
   Compiling fixed v1.16.0
error[E0658]: trait bounds other than `Sized` on const fn parameters are unstable
   --> /Users/alex/.cargo/registry/src/github.com-1ecc6299db9ec823/fixed-1.16.0/src/macros_no_frac.rs:30:14
    |
30  |           impl<Frac> $Fixed<Frac> {
    |                ^^^^
...
763 | /                     pub const fn wide_mul<RhsFrac>(
764 | |                         self,
765 | |                         rhs: $Fixed<RhsFrac>,
766 | |                     ) -> $Double<Sum<Frac, RhsFrac>>
767 | |                     where
768 | |                         Frac: Add<RhsFrac>,
    | |___________________________________________- function declared as const here
    |
   ::: /Users/alex/.cargo/registry/src/github.com-1ecc6299db9ec823/fixed-1.16.0/src/lib.rs:576:1
    |
576 | / fixed! {
577 | |     "An eight",
578 | |     FixedU8(u8, LeEqU8, U8, U7, "8", "9", "7", "6", "5", "4"),
579 | |     1, "0x12", "0x12", "[0x12]", "[0x12]",
...   |
582 | |     True, "16", FixedU16, u16, FixedI16, i16
583 | | }
    | |_- in this macro invocation
    |
    = note: see issue #93706 <https://github.com/rust-lang/rust/issues/93706> for more information
    = note: this error originates in the macro `fixed_no_frac` (in Nightly builds, run with -Z macro-backtrace for more info)
[...]
```

**Comment by alex-schulster** [comment]
Updating Rust from 1.60 to 1.62 solved the problem. Don't really know what was happening though.
